pkg/dns/azure/dns.go

```diff
@@ -154,3 +157,19 @@ func getARecordName(recordDomain string, zoneName string) (string, error) {
 	}
 	return strings.TrimSuffix(trimmedDomain, "."+zoneName), nil
 }
+
+// GetTagList returns a list of tags by merging the OCP default tags
+// and the user-defined tags present in the Infrastructure.Status
+func GetTagList(infraStatus *configv1.InfrastructureStatus) map[string]*string {
+	tags := map[string]*string{
+		fmt.Sprintf("kubernetes.io_cluster.%v", infraStatus.InfrastructureName): to.StringPtr("owned"),
+	}
+	if infraStatus.PlatformStatus != nil &&
+		infraStatus.PlatformStatus.Azure != nil &&
+		infraStatus.PlatformStatus.Azure.ResourceTags != nil {
+		for _, tag := range infraStatus.PlatformStatus.Azure.ResourceTags {
+			tags[tag.Key] = to.StringPtr(tag.Value)
```

Comment:
Do you want to make sure that the "kubernetes.io_cluster.*" field cannot be accidentally overwritten by these resource tags?  Just in case someone thought this field was supposed to be configurable.

Reply:
A check is present in installer to not allow any user defined tag with kubernetes.io prefix and validation is added in Infrastructure CRD to make tags immutable.